### PR TITLE
Add AI-Ops analytics project - Openshift SME

### PR DESCRIPTION
**Pull request by tumido**
## Related Issues

https://github.com/aicoe-aiops/openshift-sme-mailing-list-analysis/issues/2
Requires: https://github.com/AICoE/aicoe-cd/pull/45

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Adds Openshift SME App to `dh-stage-argo`. It's a private repo. **HELP wanted**

## Checklist for migrating an Application to ArgoCD

When migrating an application's deployment to be managed by ArgoCD use the following checklist to verify your process.

Items to complete before making a migration PR
- [x] Ensure your application manifests can be built using Kustomize.
- [x] If using secrets, make sure to include the .sops.yaml file in your repository.
- [x] Make sure to create the role granting access to namespace. See [here](cluster_ns_management.md#deploying-to-a-namespace) for more info. This role is tracked in your application manifests.

Items to include within the migration PR
- [x] Ensure your namespace exists in the cluster spec in the [prod-vars.yaml](../vars/prod-vars.yaml) file (for the appropriate cluster).
- [x] If you are switching between ArgoCD managed namespaces, and that namespace was deleted in OCP, then ensure it is also removed from [prod-vars.yaml](../vars/prod-vars.yaml).
- [x] Ensure the application repository is added in the `argo_cm` file in [prod-vars.yaml](../vars/prod-vars.yaml).
- [x] Ensure that all OCP resources that will be managed by ArgoCD on this cluster are included in the `inclusions` list in [prod-vars.yaml](../vars/prod-vars.yaml). See [here](cluster_ns_management.md#cluster-inclusions) for more info.
- [x] Create the ArgoCD Application, see [here](application_management.md) for instructions.

### CODE REVIEW

**Comment by tumido** [comment]
Adding Openshift SME means adding a private repo to the Argo CD. Can we use `sesheta` ssh key to access this repo?

**Comment by tumido** [comment]
@goern @HumairAK @harshad16 any idea how can we integrate Argo CD and sesheta to pull from private repos? Sesheta is already member of the org, so I think it would be the easiest way to access the private repos via this bot.

If it's already available and I've missed it, how can I use it?

**Comment by goern** [comment]
For thoth-station we have created a SourceOps team, that has admin access to all the repos, Sesheta is a member of it, so it can access all areas. Using the GitHub Access token, it can even reconfigure repo settings. 

I propose the same structure for AICoE, so that Sesheta is able to get/write/admin all repos the SourceOps team is admin for.

Argo should be able to use the GitHub token to access the repo, or we could create a r/o access token for argo.

**Comment by HumairAK** [comment]
Yep access token works: https://argoproj.github.io/argo-cd/user-guide/private-repositories/#access-token 

Encrypt using sops, instructions here: https://github.com/AICoE/aicoe-cd/blob/master/docs/secret_management.md

**Comment by tumido** [comment]
@goern can you or anybody on your team please create a secret in this repo with Sesheta token and share the secret name/key here, so I can reference it in the repo spec?

**Comment by tumido** [comment]
blocked, because Argo CD can't pull from private repos. Can be fixed by either #50 or #56 (preferred). Please hold until #56 is merged.

/hold

**Comment by tumido** [comment]
@HumairAK I've rebased this on top of the new kustomize layout + re-added the `srcops` secret using the sesheta token. Please LMKWYT

@anishasthana @harshad16 please review

File: manifests/overlays/dev/secrets/auth/secret-generator.yaml [hunk 1]
```diff
@@ -5,3 +5,4 @@ metadata:
 files:
 - secrets/auth/argocd-secret.enc.yaml
 - secrets/auth/dex-server-sa.enc.yaml
+- secrets/auth/srcops.enc.yaml
```
Comment (HumairAK) [inline manifests/overlays/dev/secrets/auth/secret-generator.yaml:8]: hrmm why do we need this for the dev env?

Comment (tumido) [reply manifests/overlays/dev/secrets/auth/secret-generator.yaml:8]: No reason right now probably. My reasoning was that we may want to have the same credentials available in both envs, so it behaves the same, no matter where you put the repo... I admit there's no usecase for it now. Do you want me to remove it?

Comment (HumairAK) [reply manifests/overlays/dev/secrets/auth/secret-generator.yaml:8]: no I think that's a fair reason, lgtm

**Pull request merged**